### PR TITLE
Use notebookIdentity instead of the notebook when handling external buttons

### PR DESCRIPTION
**Pull request by DavidKutu**
For #414

It handles the case when the user doesn't autostart the kernel.

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: pythonFiles/pyvsc-run-isolated.py [hunk 1]
```diff
@@ -4,14 +4,20 @@
...
-# on to runpy.
-sys.path[0] = os.path.join(os.path.dirname(__file__), ".does-not-exist")
+
+def normalize(path):
+    return os.path.normcase(os.path.normpath(path))
+
+
+# We "isolate" the script/module (sys.argv[1]) by removing current working
+# directory or '' in sys.path and then sending the target on to runpy.
+cwd = normalize(os.getcwd())
```
Comment (IanMatthewHuff) [inline pythonFiles/pyvsc-run-isolated.py:19]: I feel like I'm missing context on the run-isolated change. What is this for?

Comment (DavidKutu) [reply pythonFiles/pyvsc-run-isolated.py:19]: Nothing related. I was hitting an error related to executing 'import ipykernel'. I updated this file while trying to fix that. And while that wasn't it, I don't see why not keep this change, it just normalizes paths.

Comment (DonJayamanne) [reply pythonFiles/pyvsc-run-isolated.py:19]: Personally I would prefer to have a separate PR for such changes.
Helps with simpler reviews, track changes & revert commits as well.

**Review by DonJayamanne — APPROVED** [review]

File: pythonFiles/pyvsc-run-isolated.py (hunk 1 above)
Comment (IanMatthewHuff) [reply pythonFiles/pyvsc-run-isolated.py:19]: Yup, this run-isolated file can get pretty funky in some cases, so I think we should consider this as a separate fix. But the interactiveBase part all looks good to me.

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**